src/charm.py

```diff
@@ -156,21 +143,15 @@ def _on_start(self, event: ops.charm.StartEvent) -> None:
             return
 
         # do not initialise replica set until all peers have joined
-        if not self._number_of_expected_units == self._number_of_current_units:
-            logger.debug(
-                "planned units does not match joined units: %s != %s,"
-                + "waiting to initialise replica set",
-                str(self._number_of_expected_units),
-                str(self._number_of_current_units),
-            )
+        if not self._check_unit_count:
+            logger.debug("waiting to initialise replica set until all planned units have joined")
             return
 
         # initialise replica set
         try:
             self._initialise_replica_set()
         except (ConnectionFailure, ConfigurationError):
-            self.unit.status = BlockedStatus(
-                "failed to initialise replica set")
+            self.unit.status = BlockedStatus("failed to initialise replica set")
```

Comment:
Hi Jon! I initially had this as part of the `_initialise_replica_set` function. But decided to pull it out so that I could return on the condition that it throws an exception on [line 74](https://github.com/canonical/mongodb-operator/blob/21dc4c51036006f84110f8bc97f943c5b8a99b43/src/charm.py#L74) before checking if the replica set is ready [lines 77-79](https://github.com/canonical/mongodb-operator/blob/21dc4c51036006f84110f8bc97f943c5b8a99b43/src/charm.py#L76-L79). I am happy to address this now, or in my subsequent PR(s) where I refactor the replication code. Let me know what you prefer. :)

Reply:
A subsequent PR as part of your refactoring is fine!